internal/commands/list.go

```diff
@@ -95,3 +83,31 @@ func printValues(w io.Writer, installation *store.Installation) {
 	}
 	fmt.Fprintln(w, strings.Join(values, "\t"))
 }
+
+func getInstallations(targetContext, configDir string) ([]*store.Installation, error) {
+	appstore, err := store.NewApplicationStore(configDir)
+	if err != nil {
+		return nil, err
+	}
+	installationStore, err := appstore.InstallationStore(targetContext)
+	if err != nil {
+		return nil, err
+	}
+	installationNames, err := installationStore.List()
+	if err != nil {
+		return nil, err
+	}
+	installations := make([]*store.Installation, len(installationNames))
+	for i, name := range installationNames {
+		installation, err := installationStore.Read(name)
+		if err != nil {
+			return nil, err
+		}
+		installations[i] = installation
```

Comment:
Nit, could be:
```go
installations := make([]*store.Installation, 0, len(installationNames))
for ....  {
...
     installations = append(installations, installation)
```
but makes no difference I guess.

Reply:
The 2 common patterns I usually see are:
1: you don't know the size (element are filtered during iteration), and you use append
2: you know the exact size so you can pre-make the slice and directly set each element